Tiltfile

```
@@ -387,6 +391,106 @@ def prepare_all():
     )
     local(cmd, env = settings.get("kustomize_substitutions", {}))
 
+# create flavor resources from cluster-template files in the templates directory
+def flavors():
+    substitutions = settings.get("kustomize_substitutions", {})
+
+    flavor_dirs = settings.get("flavor_dirs", {
+        "docker": ["./test/infrastructure/docker/templates"],
+    })
+
+    for provider, provider_dirs in flavor_dirs.items():
+        for flavor_dir in provider_dirs:
+            template_list = [item for item in listdir(flavor_dir)]
+            template_list = [template_path for template_path in template_list if os.path.basename(template_path).endswith("yaml")]
+
+            for template_path in template_list:
+                deploy_templates(template_path, provider, substitutions)
+
+def deploy_templates(template_path, provider, substitutions):
+    # validate template_path exists
+    if not os.path.exists(template_path):
+        fail(template_path + " not found")
+
+    os.environ["NAMESPACE"] = substitutions.get("NAMESPACE", "default")
+    os.environ["KUBERNETES_VERSION"] = substitutions.get("KUBERNETES_VERSION", "v1.22.6")
+    os.environ["CONTROL_PLANE_MACHINE_COUNT"] = substitutions.get("CONTROL_PLANE_MACHINE_COUNT", "1")
+    os.environ["WORKER_MACHINE_COUNT"] = substitutions.get("WORKER_MACHINE_COUNT", "3")
+
+    basename = os.path.basename(template_path)
+    if basename.endswith(".yaml"):
+        if basename.startswith("clusterclass-"):
+            flavor = basename.replace("clusterclass-", "").replace(".yaml", "")
+            deploy_clusterclass_template(flavor, provider, template_path)
+        elif basename.startswith("cluster-template-"):
+            flavor = basename.replace("cluster-template-", "").replace(".yaml", "")
+            deploy_flavor_template(flavor, provider, template_path)
+
+def deploy_clusterclass_template(flavor, provider, template_path):
```

```suggestion
def deploy_clusterclass(name, provider, template_path):
```
ClusterClass is not really a template, even though clusterctl runs templating over it, but that's not an intended feature